src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -0,0 +1,85 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using CarbonAware.DataSources.Configuration;
+using CarbonAware.WebApi.IntegrationTests;
+using NUnit.Framework;
+using System.Net;
+using System.Web;
+
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture(DataSourceType.JSON)]
+[TestFixture(DataSourceType.WattTime)]
+public class CarbonAwareControllerTests : IntegrationTestingBase
+{
+    private string healthURI = "/health";
+    private string fakeURI = "/fake-endpoint";
+    private string bestLocationsURI = "/emissions/bylocations/best";
+
+    public CarbonAwareControllerTests(DataSourceType dataSource) : base(dataSource) { }
+
+    [Test]
+    public async Task HealthCheck_ReturnsOK()
+    {
+        //Use client to get endpoint
+        var result = await _client.GetAsync(healthURI);
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+    }
+
+    [Test]
+    public async Task FakeEndPoint_ReturnsNotFound()
+    {
+        var result = await _client.GetAsync(fakeURI);
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+    }
+
+    //ISO8601: YYYY-MM-DD
+    [TestCase("2022-1-1", 1, "eastus")]
+    [TestCase("2021-12-25", 1, "westus")]
+    public async Task BestLocations_ReturnsOK(DateTime start, int offset, string location)
```

Comment:
Prefer `DateTimeOffset`

Reply:
Might be easier for the humans reading this to just use `start, end, location` as the testcase data